runtime/klee-libc/fortify-klibc.c

```diff
@@ -25,18 +22,26 @@ void *__mempcpy_chk(void *dest, const void *src, size_t len, size_t destlen) {
   return mempcpy(dest, src, len);
 }
 
+char *stpcpy(char * to, const char * from);
```

Comment:
I wonder if these signatures should contain the `restrict` keyword where appropriate?

Reply:
The `restrict` keyword has no effect on function declarations. It is only relevant inside the function body (https://llvm.org/devmtg/2021-11/slides/2021-ptr_provenanceAndLlvmNoaliasTheTaleOfFullRestrict.pdf page 3). Godbolt: https://godbolt.org/z/zf5aGbG4G

Since it doesn't make a different I can change it though 🤷🏻‍♂️ Maybe somebody will copy the declaration and implement it somewhere else...

Reply:
Could it be that a compiler that consumes the declaration will make some decisions that will result in a generated code with a slightly different behavior?

Reply:
No, the declaration is local to the translation unit and you can see the generated LLVM IR with `restrict` does not change anything about that.

Reply:
Ah, you're right, the application code will use a definition coming from the real libc.